crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -43,10 +43,14 @@ def upgrade!
         end
 
         if upgrade_script_path.exist?
+          upgrade_status = ::Crowbar::UpgradeStatus.new
+          upgrade_status.start_step
           pid = spawn("sudo #{upgrade_script_path}")
           Process.detach(pid)
           Rails.logger.info("#{upgrade_script_path} executed with pid: #{pid}")
 
+          # we can't really call upgrade_status.end_step here yet as the upgrade is running
+          # in the background
```

Comment:
It should be called by the code which finishes executing the step.  Maybe that will require the script calling the API to mark the step as done, in which case I guess we'd need a new endpoint.  Or the script could call a standalone Ruby script which uses the state persistence library to mark the step as done.